src/api/SubjectsApi.ts

```diff
@@ -49,24 +51,52 @@ function getEndpoint( type: SubjectType ): string {
 }
 
 function fromApiResponse( type: SubjectType, response: ApiPost ): Subject {
-	const date = newDateField( response.rawDate, response.parsedDate );
-	const title = newTextField( response.rawTitle, response.parsedTitle ?? '' );
-	const content = newHtmlField(
-		response.rawContent,
-		response.parsedContent ?? ''
-	);
+	const schema = getSchema( type );
+
+	const fields = Object.entries( schema.fields ).reduce<
+		Record< string, Field >
+	>( ( acc, [ fieldName, schemaField ] ) => {
+		// Create the raw/parsed key names from the field name
+		const rawKey = `raw${ fieldName
+			.charAt( 0 )
+			.toUpperCase() }${ fieldName.slice( 1 ) }`;
+		const parsedKey = `parsed${ fieldName
+			.charAt( 0 )
+			.toUpperCase() }${ fieldName.slice( 1 ) }`;
+
+		// Get values from response
+		const rawValue = response[ rawKey ];
+		const parsedValue = response[ parsedKey ];
+
+		// Create field based on schema-defined type
+		switch ( schemaField.type ) {
+			case FieldType.Date:
+				acc[ fieldName ] = newDateField( rawValue, parsedValue );
+				break;
+			case FieldType.Text:
+				acc[ fieldName ] = newTextField( rawValue, parsedValue );
+				break;
+			case FieldType.Html:
+				acc[ fieldName ] = newHtmlField( rawValue, parsedValue );
+				break;
+			case FieldType.Link:
+				acc[ fieldName ] = newLinkField(
+					rawValue,
+					...( parsedValue.split( '||' ) as [ string, string ] )
```

Comment:
@psrpinto The only way I could think of supplying link's parsed value in a single string was to use a separator like `||`. Let me know if you have ideas around handling this better.

Reply:
I'm not sure what the correct solution here would be. For the moment, I think we could postpone this decision by not handling `link` here (removing the `case` for it) as there will be no field coming from the API of type `link` yet. At the moment, `LinkField` is only used in the UI (to parse navigation), but it's not yet sent to the server (as there is no need to store it yet).

Once we add `link` as a [supported type of field in the schema](https://github.com/WordPress/try-wordpress/blob/trunk/schema/meta/schema.json#L10-L12), we would come back to this and figure out what to do. Would that make sense to you?

Reply:
Sure, I will comment out this much code rather than just deleting it.

Reply:
I'm not sure I see the value in keeping this code around, as it's a simple call to `newLinkField()`, so I think we'd be able to write the code again easily. I also have concerns about the specific implementation (using `||`), as to me that is a sign that there's something off in the way we're expressing the data model. I think if we need to resort to this sort of "hack", something must be off in the data model.

That's why I was suggesting to just remove it, because I think that properly supporting `LinkField` (and other fields that are not just a string) will require re-evaluating the approach we're currently using for sending fields across the API.

If you think there is value in keeping this code commented out, I'm ok with it, but I generally don't see the value in keeping commented out code in repos.

Reply:
> keeping this commented code is to serve as a pointer to come back to this exact discussion

I didn't understand your intention initially, I'm fine with this. In that case, would it make sense to add a comment in the code with a link to this discussion?

Reply:
Could have explicitly linked the PR instead indeed, but merged it already now.